src/target/at32f43x.c

```diff
@@ -0,0 +1,376 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+ * Written by ALTracer <tolstov_den@mail.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file implements AT32F43x target specific functions for detecting
+ * the device, providing the XML memory map and Flash memory programming.
+ *
+ * References:
+ * ARTERY doc - RM_AT32F435_437_EN_V2.04.pdf
+ *   Reference manual - AT32F435/437 Series Reference Manual
+ */
+
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+
+static bool at32f43_flash_erase(target_flash_s *flash, target_addr_t addr, size_t len);
+static bool at32f43_flash_write(target_flash_s *flash, target_addr_t dest, const void *src, size_t len);
+static bool at32f43_mass_erase(target_s *target);
+
+/* Flash memory controller register map */
+#define AT32F435_FLASH_REG_BASE 0x40023c00U
+#define AT32F435_FLASH_UNLOCK   (AT32F435_FLASH_REG_BASE + 0x04U)
+#define AT32F435_FLASH_STS      (AT32F435_FLASH_REG_BASE + 0x0cU)
+#define AT32F435_FLASH_CTRL     (AT32F435_FLASH_REG_BASE + 0x10U)
+#define AT32F435_FLASH_ADDR     (AT32F435_FLASH_REG_BASE + 0x14U)
+//#define AT32F435_FLASH_UNLOCK2  (AT32F435_FLASH_REG_BASE + 0x44U)
+//#define AT32F435_FLASH_STS2     (AT32F435_FLASH_REG_BASE + 0x4cU)
+//#define AT32F435_FLASH_CTRL2    (AT32F435_FLASH_REG_BASE + 0x50U)
+//#define AT32F435_FLASH_ADDR2    (AT32F435_FLASH_REG_BASE + 0x54U)
+
+#define FLASH_BANK1_REG_OFFSET 0x00U
```

Comment:
Ideally all these defines want prefixing with `AT32F43_`, and the 5 above can have the 5 dropped or changed to an x in light of the file name

Reply:
These two are used in 6 & 7 places, am I to rename them?
Does this apply to all the bitfield macros, too? It may introduce more Very Long lines, as the bits are usually used in calls with long (macro) arguments and at some level of nesting/indentation.
`AT32F435` + `AT32F437` = `AT32F43X` or lowercase `AT32F43x`?

Reply:
Yes, the review item is to rename all in-file macros that define things about the devices' registers to be prefixed with `AT32F43x` and, yes, with the lower-case x there (it's consistent with targets like LPC43xx).

We're not worried about long lines as clang-format will reflow them to keep them <= 120 characters long. Being properly descriptive matters more to us.